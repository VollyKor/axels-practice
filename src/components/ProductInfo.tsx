import {
    ProductDescr,
    ProductName,
    ProductQn,
    ProductsContainer,
} from 'styled/ProductInfo';

interface Props {
    name?: string;
    color?: string;
    quantity?: number;
}

const ProductInfo = ({
    name = 'unknown',
    color = 'default',
    quantity = 1,
}: Props) => (
    <ProductsContainer className="p-1">
        <ProductName className="h6">{name}</ProductName>
        <ProductDescr>{color}</ProductDescr>
        <ProductQn>Quantity {quantity}</ProductQn>
    </ProductsContainer>
);

export default ProductInfo;
