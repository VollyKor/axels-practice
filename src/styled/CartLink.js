import styled from 'styled-components';
import { IoCartOutline } from 'react-icons/io5';

import { Wrapper } from './utils';

export const Cart = () => (
    <Wrapper>
        <span className="mr-1">cart</span>
        <IoCartOutline size="25px" />
        <p>1</p>
    </Wrapper>
);

export const CartWrapper = styled.div`
    position: relative;
    color: var(--cart-link);
`;
export const CartText = styled.span``;

export const CartNotification = styled.div`
    position: absolute;
    top: 0;
    right: -2px;

    width: 12px;
    height: 12px;

    color: var(--white-color);
    background-color: var(--pink-color);
    border-radius: 50%50%;

    font-size: 9px;
`;
