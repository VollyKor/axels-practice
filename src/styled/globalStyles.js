import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
:root{
  /* bootstrap variable */
  --primary :  #5f2093;

  /* app variables */
  --main-bg-color: #E6E9F0;
  --accent-bg-color: #F1F3F6;

  --main-text-color: #5B1195;
  --accent-text-color : #808181;

  --cart-link: #c90ea5;

  --black-color: #212121;
  --pink-color: #ff1876;

  /* doesn't work */
  --breakpoint-xs: 0px;
  --breakpoint-sm: 576px;
  --breakpoint-md: 768px;
  --breakpoint-lg: 992px;
  --breakpoint-xl: 1200px;
  --breakpoint-xxl: 1400px;
}

  body {
    box-sizing: border-box;

    margin: 0;
    padding: 0;

    background-color: var(--main-bg-color);

    font-family: Open-Sans, Helvetica, Sans-Serif;
    color: var(--main-text-color);
  }

  a {
    color: inherit
    }
`;

export default GlobalStyle;

// --primary  main-text-color: #5B1195;
// accent-text-color : $e6e9f0
// muted-color:
// black-color: $2f2929;
// white-color
