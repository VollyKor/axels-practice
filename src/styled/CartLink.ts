import styled from 'styled-components';

export const CartName = styled.span``;
export const CartQuantyty = styled.p``;

export const CartWrapper = styled.div`
    position: relative;
    color: var(--cart-link);
`;
export const CartText = styled.span``;

export const CartNotification = styled.div`
    position: absolute;
    top: 0;
    right: -2px;

    width: 12px;
    height: 12px;

    color: var(--white-color);
    background-color: var(--pink-color);
    border-radius: 50%50%;

    font-size: 9px;
`;
