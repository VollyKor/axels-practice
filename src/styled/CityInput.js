import styled from 'styled-components';

export const FindButton = styled.button`
    position: absolute;
    z-index: 1;

    right: 0;
    top: 4px;

    background-color: transparent;
    border: none;
`;
