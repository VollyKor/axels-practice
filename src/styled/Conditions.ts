import styled from 'styled-components';

export const ConditionText = styled.p`
    bottom: 0;
    position: initial;
    text-align: center;
    font-size: 0.6rem;

    @media ${({ theme }) => theme?.deviceSize?.conditionTextWidth} {
        position: absolute;
        margin-top: 0.8rem;
    } ;
`;

export const ConditionLink = styled.a`
    font-size: 0.65rem;
    color: var(--text-accent-color);
`;
