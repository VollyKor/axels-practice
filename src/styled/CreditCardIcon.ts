import styled from 'styled-components';

export const IconWrapper = styled.div`
    position: absolute;
    top: 8px;
    right: 5px;
`;

export const Icon = styled.img`
    display: block;

    height: 24px;
`;
