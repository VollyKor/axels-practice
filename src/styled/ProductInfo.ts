import styled from 'styled-components';

export const ProductsContainer = styled.div``;

export const ProductName = styled.h2`
    margin-bottom: 2px;
    color: var(--text-accent-color);
    font-size: 0.8rem;
`;

export const ProductDescr = styled.p`
    margin-bottom: 0;
    color: var(--black-color);

    font-size: 0.65rem;
`;

export const ProductQn = styled(ProductDescr)``;
