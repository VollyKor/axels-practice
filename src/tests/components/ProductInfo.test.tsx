import { shallow } from 'enzyme';

import ProductInfo from 'components/ProductInfo';
import { ProductDescr, ProductName, ProductQn } from 'styled/ProductInfo';

it('renders snapshot', async () => {
    const wrapper = shallow(<ProductInfo />);
    expect(wrapper).toMatchSnapshot();
});

it('render props correctly', async () => {
    const testName = 'bob';
    const testColor = 'red';
    const testQty = 1;

    const wrapper = shallow(
        <ProductInfo name={testName} color={testColor} quantity={testQty} />
    );

    const name = wrapper.find(ProductName).text();
    const color = wrapper.find(ProductDescr).text();
    const qty = wrapper.find(ProductQn).text();

    expect(name).toEqual(testName);
    expect(color).toEqual(testColor);
    expect(qty).toEqual(`Quantity ${testQty}`);
});
it('render default props correctly', async () => {
    const wrapper = shallow(<ProductInfo />);

    const name = wrapper.find(ProductName).text();
    const color = wrapper.find(ProductDescr).text();
    const qty = wrapper.find(ProductQn).text();

    expect(name).toEqual('unknown');
    expect(color).toEqual('default');
    expect(qty).toEqual(`Quantity 1`);
});
