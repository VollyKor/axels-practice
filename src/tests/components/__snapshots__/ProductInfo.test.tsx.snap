// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<styled.div
  className="p-1"
>
  <styled.h2
    className="h6"
  >
    unknown
  </styled.h2>
  <styled.p>
    default
  </styled.p>
  <Styled(styled.p)>
    Quantity 
    1
  </Styled(styled.p)>
</styled.div>
`;
